File: year2020/month09/day0922/ms_0922.cpp
#include <iostream>
#include <cmath>

#define SQUARE(x, y) pow((x - y), 2)
#define CUBE(x, y) pow((x + y), 3)

using namespace std;

int main() {
	int x, y;

	cin >> x >> y;

	cout << "(" << x << " - " << y << ") ^ 2 = " << SQUARE(x, y) << endl;
	cout << "(" << x << " + " << y << ") ^ 3 = " << CUBE(x, y) << endl;

	return 0;
}

File: year2020/month09/day0922/ms_0922.cs
using System;

namespace algorithm_62
{
    delegate double DelegateMethod(int x, int y, int count);

    class ms_0922
    {
        public static double minusSquare(int x, int y, int count) { return Math.Pow((x - y), count); }
        public static double plusCube(int x, int y, int count) { return Math.Pow((x + y), count); }
        
        static void Main(string[] args)
        {
            string[] input = Console.ReadLine().Split(' ');
            int x = Convert.ToInt32(input[0]);
            int y = Convert.ToInt32(input[1]);

            DelegateMethod square = new DelegateMethod(minusSquare);
            DelegateMethod cube = new DelegateMethod(plusCube);

            Console.WriteLine($"({x} - {y}) ^ 2 = {square(x, y, 2)}");
            Console.WriteLine($"({x} + {y}) ^ 3 = {cube(x, y, 3)}");
        }
    }
}
